components/Home/Epoch.tsx

```diff
@@ -289,90 +274,84 @@ const Epoch = (): JSX.Element => {
-                                    {data.epoch.epochSize}
-                                </Typography>
+                            {blockProposer?.data?.blockAdded?.number &&
+                            chain?.data?.chain?.firstBlockNumberForEpoch &&
+                            chain?.data?.chain?.epochSize ? (
+                                <>
+                                    <Typography variant="body1" noWrap>
+                                        {blockProposer?.data?.blockAdded?.number -
+                                            chain?.data?.chain?.firstBlockNumberForEpoch >
+                                        0
+                                            ? blockProposer?.data?.blockAdded?.number -
+                                              chain?.data?.chain?.firstBlockNumberForEpoch
+                                            : 0}
+                                    </Typography>
+
+                                    <Divider variant="middle" className={classes.divider} />
+
+                                    <Typography variant="body1" noWrap>
+                                        {chain?.data?.chain?.epochSize}
+                                    </Typography>
+                                </>
                             ) : null}
                         </Grid>
-                        {blockProposer &&
-                        blockProposer.data &&
-                        blockProposer.data.blockAdded &&
-                        blockProposer.data.blockAdded.number &&
-                        blockProposer.data.blockAdded.miner ? (
-                            <Grid item xs={12} className={classes.blockProposer}>
-                                <img
-                                    src={`https://ui-avatars.com/api/?rounded=true&size=40&name=${blockProposer.data.blockAdded.miner.name}&color=rgba(8, 178, 122, 1)&background=fff`}
-                                    className={classes.roundIcon}
-                                    alt="Block Proposer"
-                                />
-                                {blockProposer.data.blockAdded.miner.name ? (
-                                    <Typography
-                                        variant="body2"
-                                        color="textPrimary"
-                                        className={classes.blockProposerName}>
-                                        {blockProposer.data.blockAdded.miner.name}
-                                    </Typography>
-                                ) : (
+                        <Grid item xs={12} className={classes.blockProposer}>
+                            {blockProposer?.data?.blockAdded?.miner?.name ||
+                            blockProposer?.data?.blockAdded?.miner?.address ? (
+                                <>
+                                    <img
+                                        src={`https://ui-avatars.com/api/?rounded=true&size=40&name=${
```

Comment:
How about we switch this to https://jdenticon.com/ ?

Reply:
Ok, I added Avatar component and switched to using jdenticon library to generate icons :)